sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ApplicationInsightsSampler.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter
+{
+    public class ApplicationInsightsSampler : Sampler
+    {
+        private readonly float samplingRatio;
+
+        public ApplicationInsightsSampler(float samplingRatio)
+        {
+            // Ensure passed ratio is between 0 and 1, inclusive
+            if (samplingRatio < 0 || samplingRatio > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(samplingRatio), "Ratio must be between 0 and 1, inclusive.");
+            }
+            this.samplingRatio = samplingRatio;
+            Description = "ApplicationInsightsSampler{" + samplingRatio + "}";
+        }
+
+        public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+        {
+            double sampleScore = DJB2SampleScore(samplingParameters.TraceId.ToHexString());
```

Comment:
We might need to convert this to lowercase since the `ToHextString` API [doc](https://docs.microsoft.com/dotnet/api/system.diagnostics.activitytraceid.tohexstring?view=net-5.0) doesn't say anything about the upper/lower case (which means certain implementation might choose to return uppercase).

Reply:
Yep makes sense, implemented, using ToLowerInvariant() instead of ToLower()